.github/workflows/node-ci.yml

```diff
@@ -11,17 +11,25 @@ on:
     - master
     - feature/*
 
-
+# Our jobs run like this to minimize wasting resource cycles:
+#   1. Prime caches for primary configuration (ubuntu on node 12).
+#      This way the next two jobs can run in parallel but rely on this primed
+#      cache.
+#   2. Lint, coverage, and publish
+#     a. Lint
+#     b. Coverage and publish
+#   3. Run tests for remaining configurations.
+#      Since these don't share caches, we don't need to prime those caches.
 jobs:
-  prime_cache:
-    name: Prime node_modules cache
+  prime_cache_primary:
+    name: Prime node_modules cache for primary configuration
     env:
       CI: true
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macOS-latest]
-        node-version: [12.x, 14.x]
+        os: [ubuntu-latest]
+        node-version: [12.x]
```

Comment:
We only bother priming our primary platform now. This is because the lint job and the publish job run side-by-side and so priming the cache saves them time. However, the test runs that follow the publish and lint jobs are independent of one another and therefore don't benefit from priming at all.